model/pipeline/src/main/proto/metrics.proto

```diff
@@ -229,101 +215,127 @@ message MonitoringInfo {
     NAMESPACE = 5 [(label_props) = { name: "NAMESPACE" }];
     NAME = 6 [(label_props) = { name: "NAME" }];
   }
+
   // A set of key+value labels which define the scope of the metric.
   // Either a well defined entity id for matching the enum names in
   // the MonitoringInfoLabels enum or any arbitrary label
   // set by a custom metric or user metric.
+  //
   // A monitoring system is expected to be able to aggregate the metrics
   // together for all updates having the same URN and labels. Some systems such
   // as Stackdriver will be able to aggregate the metrics using a subset of the
   // provided labels
-  map<string, string> labels = 5;
-
-  // The walltime of the most recent update.
-  // Useful for aggregation for latest types such as LatestInt64.
-  google.protobuf.Timestamp timestamp = 6;
+  map<string, string> labels = 4;
 }
 
 message MonitoringInfoTypeUrns {
   enum Enum {
+    // Represents an integer counter where values are summed across bundles.
+    //
+    // Encoding: <value>
+    //   - value: beam:coder:varint:v1
     SUM_INT64_TYPE = 0 [(org.apache.beam.model.pipeline.v1.beam_urn) =
```

Comment:
I think there are only a few of these types being used now. SUM_INT64_TYPE and DISTRIBUTION_INT64_TYPE. I hope we can make it very simple to add new ones of these with minimal changes (Adding MonitoringInfoSpec and reusing existing framework/libraries in the SDK, runners can mostly pass them through to a service to aggregate across multiple workers)

Reply:
The problematic type is `string`:
Is `sum` the same as concat? (if so, this will grow to be huge very quickly also making distribution pointless)
Is min/max/top_n/bottom_n based upon lexicographical ordering? (if so, case sensitive/insensitive/...)

`string` only makes sense with latest and if we wanted to define any of the other ones I think it would be better to use terms like `case_insensitive_max`.

Reply:
Agree that string summation (and those metrics requiring it) could be omitted, but min, max, top_n, bottom_n are meaningful. (And distribution could still be meaningful giving a count and range.)